### ISSUE TITLE
100% CPU constant usage on "Data Decoder" and "GPU"

### ISSUE THREAD

**Issue by rohitsaha29** [Alex313031/Mercury#226]
Windows 11 Pro 23H2
Intel i7-13700k & 2080ti

Version: [v.129.0.2] AVX2 and AVX

Hardware Acceleration is forced and on, stock no custom user js or anything

![image](https://github.com/user-attachments/assets/108466c7-fa4f-4f66-85a5-b4459cb4ed31)

**Comment by gz83** [comment]
Isn't this the expected behavior with HW turned on?

**Comment by rohitsaha29** [comment]
> Isn't this the expected behavior with HW turned on?

No it is not, i have a 100% cpu usage when I have HW acell on. This doesnt happen on stock firefox or any other fork of firefox. This even happens on the normal non AVX build of mercury.

**Comment by gz83** [comment]
Sorry, I didn't read your description clearly when I was typing that comment a few hours ago on my phone.

Can you confirm that you have installed the HEVC plugin from the Microsoft Store?

**Comment by rohitsaha29** [comment]
> Sorry, I didn't read your description clearly when I was typing that comment a few hours ago on my phone.
> 
> Can you confirm that you have installed the HEVC plugin from the Microsoft Store?

Yes I do have it installed.

**Comment by Soromeister** [comment]
@rohitsaha29 Do you have Intel CPU with Intel Graphics and an NVidia GPU?

**Comment by rohitsaha29** [comment]
@Soromeister yes i do actually. is there something i need to change or do

**Comment by rohitsaha29** [comment]
has anyone found or know the fix for this please?

**Comment by Soromeister** [comment]
@rohitsaha29 I managed to fix this following the below:
- Go to `about:config` and set these values:
```
dom.webgpu.enabled = true
dom.webgpu.indirect-dispatch.enabled = true
dom.webgpu.workers.enabled = true
gfx.vsync.force-disable-waitforvblank = true
gfx.webrender.all = true
gfx.webrender.dcomp-video-hw-overlay-win-force-enabled = true
gfx.webrender.triple-buffering.enabled = false
media.hardware-video-decoding.force-enabled = true
security.sandbox.gpu.level = 0
```
- After doing this, please make sure you close all Mercury processes;
- In your Windows 11, go to `Settings > System > Display > Graphics` and find Mercury there. If it's not there, add it. Make sure that `GPU Preference` is set to `High Performance` with your NVidia GPU listed there.
- Open `NVidia Control Panel`, go to `3D Settings > Manage 3D Settings > Program Settings` and select Mercury from the list, or simply add it via the "Add" button. Once Mercury is added / selected, set it to run using `High Performance NVIDIA Processor` and, from the list below, set  `Power Management Mode` to `Adaptive`. Click on Apply to apply the settings.
- Preferably, to make sure this gets applied, reboot your computer and then start Mercury. Using Task Manager, you can monitor which GPU is being used (`GPU 0` is Intel, `GPU 1` is NVidia). Mercury should now show up as using `GPU 1` and you shouldn't get the issue anymore.

**Comment by rohitsaha29** [comment]
@Soromeister unfortunately this did not work for me. I think its just a issue with mercury itself because my friend was able to replicate it on his system and the same happened for another PC of mine. and this issue doesn't happen on FF Dev or Stable. Hopefully Alex is aware of this issue and can fix it in the next release.

**Comment by Soromeister** [comment]
@rohitsaha29 I managed to fix this following the below:
- Go to `about:config` and set these values:
```
dom.webgpu.enabled = true
dom.webgpu.indirect-dispatch.enabled = true
dom.webgpu.workers.enabled = true
gfx.vsync.force-disable-waitforvblank = true
gfx.webrender.all = true
gfx.webrender.dcomp-video-hw-overlay-win-force-enabled = true
gfx.webrender.triple-buffering.enabled = false
media.hardware-video-decoding.force-enabled = true
security.sandbox.gpu.level = 0
```
- After doing this, please make sure you close all Mercury processes;
- In your Windows 11, go to `Settings > System > Display > Graphics` and find Mercury there. If it's not there, add it. Make sure that `GPU Preference` is set to `High Performance` with your NVidia GPU listed there.
- Open `NVidia Control Panel`, go to `3D Settings > Manage 3D Settings > Program Settings` and select Mercury from the list, or simply add it via the "Add" button. Once Mercury is added / selected, set it to run using `High Performance NVIDIA Processor` and, from the list below, set  `Power Management Mode` to `Adaptive`. Click on Apply to apply the settings.
- Preferably, to make sure this gets applied, reboot your computer and then start Mercury. Using Task Manager, you can monitor which GPU is being used (`GPU 0` is Intel, `GPU 1` is NVidia). Mercury should now show up as using `GPU 1` and you shouldn't get the issue anymore.

Hope this works for you too.

**Comment by Soromeister** [comment]
Technically, the `security.sandbox.gpu.level = 0` should be enough to fix 100% GPU usage. Not sure about that "Data Decoder".

**Comment by rohitsaha29** [comment]
@Soromeister yeah i don't get it on GPU anymore but I get 100-150% on CPU usage for Data Decoder after all the thing you mentioned applied. 
![{D652C9F6-4DD0-47AE-8A39-6AB08116DFAC}](https://github.com/user-attachments/assets/a8812a4f-3e26-43a8-a994-329b47a00164)

**Comment by inteliboy** [comment]
Hi all, this has been covered serval times already. This issue is caused by the name of the Mercury executable and NVIDIA having mercury.exe as an excutable for Gotham Knights game which causes high GPU usage as NVIDIA thinks a game is running. 
https://github.com/Alex313031/Mercury/issues/116#issuecomment-2053985541

**Comment by rohitsaha29** [comment]
> Hi all, this has been covered serval times already. This issue is caused by the name of the Mercury executable and NVIDIA having mercury.exe as an excutable for Gotham Knights game which causes high GPU usage as NVIDIA thinks a game is running. [#116 (comment)](https://github.com/Alex313031/Mercury/issues/116#issuecomment-2053985541)

So is there a fix that I can do on my end? or do I have to wait for Alex to incorportate a fix?

**Comment by totherpage955** [comment]
> > Hi all, this has been covered serval times already. This issue is caused by the name of the Mercury executable and NVIDIA having mercury.exe as an excutable for Gotham Knights game which causes high GPU usage as NVIDIA thinks a game is running. [#116 (comment)](https://github.com/Alex313031/Mercury/issues/116#issuecomment-2053985541)
> 
> So is there a fix that I can do on my end? or do I have to wait for Alex to incorportate a fix?

couldn't you change the name for the exe file as a fix? or tweak nvidia's settings
